### PR TITLE
Initial commit of tsuki-tag app manifest

### PR DESCRIPTION
**Pull request by ngelsi**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: eu.ngdevelopment.tsukitag.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>eu.ngdevelopment.tsukitag</id>
+	<name>Tsuki-tag</name>
+	<summary>Anime imageboard browser, aggregator, downloader, tagger, converter, collection manager. Download images from imageboards while keeping or modifying the original tag list, browse and create your own collections, configure workspaces for customizable image downloads with file and tag templating support.</summary>
+	<metadata_license>CC-BY-SA-3.0</metadata_license>
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.appdata.xml:6]: `CC0-1.0` is prefered for metadata

File: eu.ngdevelopment.tsukitag.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			<image>https://ngelsi.ngdevelopment.eu/images/main05.jpg</image>
+		</screenshot>		
+	</screenshots>
+	<releases>
+		<release version="0.9.2" date="2022-01-28">
+			<description>
+				<p>v0.9.2 is an initial BETA release which provides the featureset described in the description above.</p>
+			</description>
+		</release>
+	</releases>
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.appdata.xml:59]: `content_rating` is missing

File: flathub.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why x86_64 only?

File: eu.ngdevelopment.tsukitag.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                },
+                {
+                    "type": "file",
+                    "path": "./icons/16x16.png",
+                    "dest-filename": "16x16.png"
+                }                
+            ]
+        }
+    ],
+    "finish-args": [
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:67]: `finish-args` and `cleanup` is usually above `modules`

File: eu.ngdevelopment.tsukitag.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        "--filesystem=xdg-pictures",
+        "--socket=x11",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--share=network"
+    ],
+    "cleanup": [
+        "rm tsukitag.tar.gz"
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:75]: 1. cleanup isn't a list of command
2. cleanup is to clean installed files not downloads or build artifacts

**Review by hfiguiere — COMMENTED** [review]
.icns and .ico are relevant and don't belong here

screenshot are never referenced so they shouldn't be in the repository either.

File: eu.ngdevelopment.tsukitag.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                {
+                    "type": "file",
+                    "url": "https://github.com/ngelsi/tsuki-tag-native/releases/download/v.0.9.2/tsukitag-linux-x64-uncompressed-v092.tar.gz",
+                    "sha256": "d9e0f22bb27a2293fdf3634b3f98631a7bc0f0227204a972c2414e1162230438",
+                    "dest-filename": "tsukitag.tar.gz"
+                },
+                {
+                    "type": "file",
+                    "path": "eu.ngdevelopment.tsukitag.desktop",
+                    "dest-filename": "eu.ngdevelopment.tsukitag.desktop"
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:32]:
```suggestion
```

File: eu.ngdevelopment.tsukitag.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+                },
+                {
+                    "type": "file",
+                    "path": "eu.ngdevelopment.tsukitag.desktop",
+                    "dest-filename": "eu.ngdevelopment.tsukitag.desktop"                    
+                },
+                {
+                    "type": "file",
+                    "path": "eu.ngdevelopment.tsukitag.appdata.xml",
+                    "dest-filename": "eu.ngdevelopment.tsukitag.appdata.xml"
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:37]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build eu.ngdevelopment.tsukitag

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/eu.ngdevelopment.tsukitag.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : <summary> is too long [Anime imageboard browser, aggregator, downloader, tagger, converter, collection manager. Download images from imageboards while keeping or modifying the original tag list, browse and create your own collections, configure workspaces for customizable image downloads with file and tag templating support.] maximum is 200 chars
Validation of files failed
```

I doubt it will run either.

**Comment by ngelsi** [comment]
I adjusted the mentioned changes.

Im a bit lost of the correct content_rating because of the nature of the application (online image browser) and tried to set it accordingly, please guide me in this adjustment if this is not adequate.

I have chosen x86_64 as the application is built on top of the Avalonia framework and compiled against x64, so i would presume that it would only work on computers using this architecture.

Thank you for your help.

File: eu.ngdevelopment.tsukitag.json [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    "app-id": "eu.ngdevelopment.tsukitag",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "TsukiTag",
+    "finish-args": [
+        "--filesystem=xdg-pictures",
+        "--socket=x11",
+        "--socket=fallback-x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:11]: does it support wayland?

Also `--share=ipc` is needed for x11 anyway.

File: eu.ngdevelopment.tsukitag.json [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        "--share=network"
+    ],
+    "cleanup": [        
+    ],
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:15]:
```suggestion
```

File: eu.ngdevelopment.tsukitag.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                "mkdir /app/bin",
+                "tar -xzf tsukitag.tar.gz -C /app/bin",
+                "install -Dm644 eu.ngdevelopment.tsukitag.appdata.xml /app/share/metainfo/eu.ngdevelopment.tsukitag.appdata.xml",
+                "install -Dm644 eu.ngdevelopment.tsukitag.desktop /app/share/applications/eu.ngdevelopment.tsukitag.desktop",
+                "install -Dm644 256x256.png /app/share/icons/hicolor/256x256/apps/eu.ngdevelopment.tsukitag.png",
+                "install -Dm644 128x128.png /app/share/icons/hicolor/128x128/apps/eu.ngdevelopment.tsukitag.png",
+                "install -Dm644 64x64.png /app/share/icons/hicolor/64x64/apps/eu.ngdevelopment.tsukitag.png",
+                "install -Dm644 32x32.png /app/share/icons/hicolor/32x32/apps/eu.ngdevelopment.tsukitag.png",
+                "install -Dm644 16x16.png /app/share/icons/hicolor/16x16/apps/eu.ngdevelopment.tsukitag.png",
+                "rm tsukitag.tar.gz"
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:30]:
```suggestion
```

File: eu.ngdevelopment.tsukitag.json (hunk 8 above)
Comment (ngelsi) [reply eu.ngdevelopment.tsukitag.json]: Yes, Wayland was tested with a Fedora vm installation from the latest ISO.

**Comment by ngelsi** [comment]
bot, build eu.ngdevelopment.tsukitag

**Comment by ngelsi** [comment]
bot, build eu.ngdevelopment.tsukitag

File: eu.ngdevelopment.tsukitag.json [hunk 11]
```diff
@@ -0,0 +1,73 @@
+{
+  "app-id": "eu.ngdevelopment.tsukitag",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "21.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "TsukiTag",
+  "finish-args": [
+    "--filesystem=xdg-pictures",
+    "--socket=x11",
+    "--socket=fallback-x11",
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:10]:
```suggestion
    "--socket=x11",
```

File: eu.ngdevelopment.tsukitag.json [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    "--share=network",
+    "--share=ipc"
+  ],
+  "cleanup": [],
```
Comment (hfiguiere) [inline eu.ngdevelopment.tsukitag.json:14]:
```suggestion
```

**Comment by barthalion** [comment]
bot, build eu.ngdevelopment.tsukitag

File: README.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
+# Tsuki-tag
```
Comment (barthalion) [inline README.md:1]: FYI there's no need to have that file here as it's unlikely to be ever seen by a user. It's just a nitpick though.

File: eu.ngdevelopment.tsukitag.appdata.xml [hunk 14]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline eu.ngdevelopment.tsukitag.appdata.xml:1]: Ideally, you should maintain it at https://github.com/ngelsi/tsuki-tag-native/.

File: eu.ngdevelopment.tsukitag.desktop [hunk 15]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
```
Comment (barthalion) [inline eu.ngdevelopment.tsukitag.desktop:1]: Ideally, you should maintain it at https://github.com/ngelsi/tsuki-tag-native/.

**Pull request closed without merging**